runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -161,6 +162,20 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 return TransactionalContext.getCurrentContext()
                         .access(this, accessMethod, conflictObject);
             } catch (Exception e) {
+                if (e instanceof TransactionAbortedException) {
+                    TransactionAbortedException tae = (TransactionAbortedException) e;
+                    if (tae.getCause() instanceof TrimmedException) {
+                        TrimmedException te = (TrimmedException) tae.getCause();
+                        AbstractTransactionalContext context = TransactionalContext.getCurrentContext();
+                        System.out.println(te.getSnapshot() + "   " + TransactionalContext
+                                .getCurrentContext().getSnapshotTimestamp());
+                        if (context.getRetry() < 1 || te.getSnapshot() != Address.NON_ADDRESS
```

Comment:
this counter is wrong. it counts retries per transaction-context, but a TX might access more than one object, and each might encounter a trimException.

The retry counter should be for the loop here only. 

Also, you don't want to retry even once if te.getSnapshot() > context.getSnapshotTS() . The retry counter should be used for two purposes:

1. bound the recursion depth
2. print a warning message if we retry twice in the loop here.